### DISCUSSION TITLE
Props destructuring defaults don't work when using webpack with Nuxt

### DISCUSSION THREAD

**Discussion by JensErikChristiansen** [nuxt/nuxt#21272] [Questions]
Hi!

With the release of Nuxt 3.5, which has Vue 3.3 in it, I have been trying out props destructuring. Destructuring props works, but unfortunately the defaults don't seem to be reactive when using Webpack as the builder.

We have to use webpack, so we can't use the following:
```JS
vite: {
  vue: {
    script: {
      propsDestructure: true,
    },
  },
}
```

Here is a little code snippet (it's not actually our code, but it doesn't work for us either way):
```
const {
  items,
  selected = 'carrot',
  onClick,
} = defineProps<{
  items: string[];
  selected?: string;
  onClick: (item: string) => void;
}>();
```

```html
<template>
  <h1>Selected: {{ selected }}</h1>
</template>
```

<img width="279" alt="image" src="https://github.com/nuxt/nuxt/assets/6833664/c66eb2ca-9175-49ec-a333-ddebf433b8cd">

Is it possible to destructure props with defaults using Webpack? Thanks in advance!

PS
When creating a new Nuxt app, and using Vite with the config above, the same code _does_ work.

**Comment by JensErikChristiansen** [comment]
Is this a bug that needs to be fixed? Or is there something I can do on my end?

**Comment by MuhammadM1998** [comment]
You're passing the `propsDestructure` to the `vite` key, that's why webpack is not picking up the change. 

Correct way as per to [docs](https://nuxt.com/docs/api/configuration/nuxt-config#propsdestructure-1) is to pass `propsDestructure` in `vue` options object as following

```javascript
export default defineNuxtConfig({
  vue: {  
    propsDestructure: true  
  },
  // ... Rest of the config
})
```

**Reply by JensErikChristiansen** [reply]
unfortunately this doesn't work. I guess I'll have to wait until propsDestructure leaves experimental stage. Thanks anyway!
